compute/test/reservations.test.js

```diff
@@ -84,6 +84,17 @@ describe('Compute reservation', async () => {
     assert.deepEqual(response, reservation);
   });
 
+  it('should update number of VMs in reservation', () => {
+    const newVMsNumber = '1';
+    const response = JSON.parse(
+      execSync('node ./reservations/reservationVmsUpdate.js', {
+        cwd,
+      })
+    );
+
+    assert.equal(response.specificReservation.count, newVMsNumber);
+  });
+
   it('should return list of reservations', () => {
```

Comment:
This test (which wasn't even modified) failed. This why I recommended in your other PR to simplify your tests. There is no need to system test production APIs in your tests (it's hard to get right and it's not a sample's responsibility; the service team exhausitively tests their services). It's only the sample's job to demonstrate a use case for an API and perhaps also demonstrate typical error handling for best practices (or comment that production code should try/catch API calls, etc). Our test code doesn't actually make into the docs, so really they are just there as a guard against breaking samples, but samples themselves ought to fully encapsulate whatever is being demonstrated. If the sample throws, it exits with an error code, which is enough to signal we have a broken sample.

<img width="652" alt="image" src="https://github.com/user-attachments/assets/ab119fab-29de-4432-b863-28e2891a1ef5">

https://github.com/GoogleCloudPlatform/nodejs-docs-samples/actions/runs/10821168316/job/30022663732?pr=3844